src/avatar/draw.js

```diff
@@ -0,0 +1,355 @@
+                matchingTextLabel.parentElement.classList.add('displayNone');
+            }
+        }
+    }
+
+    // when sending a beam, highlight your cursor
+    function renderCursorOverlay(isVisible, screenX, screenY, color) {
+        let overlay = document.getElementById('beamOverlay');
+        if (!overlay) {
+            overlay = document.createElement('div');
+            overlay.id = 'beamOverlay';
+            overlay.style.position = 'absolute';
+            overlay.style.left = '-10px';
+            overlay.style.top = '-10px';
+            overlay.style.pointerEvents = 'none';
+            overlay.style.width = '20px';
+            overlay.style.height = '20px';
+            overlay.style.borderRadius = '10px';
+            overlay.style.backgroundColor = color;
+            overlay.style.opacity = '0.5';
+            document.body.appendChild(overlay);
+        }
+        overlay.style.transform = 'translate3d(' + screenX + 'px, ' + screenY + 'px, 1201px)';
+        overlay.style.display = isVisible ? 'inline' : 'none';
+    }
+
+    // Shows an "Establishing Connection..." --> "Connected!" label in the top left
+    function renderConnectionFeedback(isConnected) {
+        if (!statusUI) {
+            statusUI = document.createElement('div');
+            statusUI.id = 'avatarStatus';
+            statusUI.style.pointerEvents = 'none';
+            statusUI.style.opacity = '0.5';
+            statusUI.style.position = 'absolute';
+            statusUI.style.width = '100vw';
+            statusUI.style.height = '100px';
+            statusUI.style.left = '5px';
+            statusUI.style.top = (realityEditor.device.environment.variables.screenTopOffset + 5) + 'px';
+            statusUI.style.zIndex = '3000';
+            statusUI.style.transform = 'translateZ(3000px)';
+            document.body.appendChild(statusUI);
+        }
+        if (hasConnectionFeedbackBeenShown) { return; }
+        if (isConnected) {
+            hasConnectionFeedbackBeenShown = true;
+            statusUI.innerText = '';
+            setTimeout(() => {
+                statusUI.innerText = 'Avatar Connected!';
+                setTimeout(() => {
+                    statusUI.innerText = '';
+                    statusUI.style.display = 'none';
+                }, 2000);
+            }, 300);
+        } else {
+            statusUI.innerText = 'Establishing Avatar Connection...'
+        }
+    }
+
+    // show some debug text fields in the top left corner of the screen to track data connections and transmission
+    function renderConnectionDebugInfo(connectionStatus, debugConnectionStatus, myId, debugMode) {
+        if (!debugMode) {
+            if (debugUI) { debugUI.style.display = 'none'; }
+            return;
+        }
+
+        if (!debugUI) {
+            debugUI = document.createElement('div');
+            debugUI.id = 'avatarConnectionStatus';
+            debugUI.style.pointerEvents = 'none';
```

Comment:
See above, having these in a css rule might even allow us to eventually standardize all the UI zIndices with a shared rule or css variable but that feels a bit far out for now